io/io/inc/TROSharedMemFile.h

```diff
@@ -0,0 +1,39 @@
+// @(#)root/io:$Id$
+// Author: Philippe Canal, May 2011
+
+/*************************************************************************
+ * Copyright (C) 1995-2009, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TROSharedMemFile
+#define ROOT_TROSharedMemFile
+
+#include "TMemFile.h"
+
+#include <memory>
+#include <vector>
+
+class TROSharedMemFile : public TMemFile {
+
+private:
+   std::shared_ptr<const std::vector<char>> fData; // shared file data / content
```

Comment:
the `shared_ptr` could be `const` since copy-assignment is private _and_ deleted. sometimes ptr constness results in compiler optimizations